### DISCUSSION TITLE
EbookUI advanced settings being ignored in 3.4.6 ?

### DISCUSSION THREAD

**Discussion by Mrnofish** [sumatrapdfreader/sumatrapdf#2950] [Q&A]
Having upgraded from an old version (3.3.3 or 4 I think), ebooks are no longer being rendered with the colors set in EbookUI, but rather with standard, printed page, black on white.

The unrecognized settings section in the advanced settings config file is empty.

Do I need to make any changes to restore the previous colors, or did I just run into a bug?

```
EbookUI [
	FontName = Georgia
	FontSize = 12
	TextColor = #c0c0c0
	BackgroundColor = #303030
	UseFixedPageUI = false
]
```

**Comment by GitHubRulesOK** [comment]
Prior to version 3.4 there were two methods to view epub (a simple one font ebook UI) and a fixed format ui based on a different method

From 3.4 those were combined into one single interface using MuPDF code (for most ebook supported formats), some previous features were impacted and some improved, so for different users the impact varies. see https://www.sumatrapdfreader.org/docs/Customizing-eBook-UI

**Comment by Mrnofish** [comment]
Thanks for the quick reply.

On a second look, the EbookUI block has indeed been moved to the bottom section "not recognized by the current version" now.

So I guess it is now impossible to have a dark theme with eBooks and a white background with PDFs ?

edit: that does seem to be the case, using a dark theme results in inverted colors in color images in a regular PDF.

> see https://www.sumatrapdfreader.org/docs/Customizing-eBook-UI

That is outdated, the EbookUI section is no longer recognized, unfortunately.

**Reply by GitHubRulesOK** [reply]
There is no theming simply press I for the all versions behavior (invert colours) If you want 2 settings for different types there were fiddles to change/reverse background/foreground colours on the fly or use two -appdata profiles with separate histories but its easier to press one key

> That is outdated,  

no it says it was changed its there for those on 3..3.3 or lower (could be clearer)  
> Note: in 3.4 we changed the eBook rendering engine and this customization is no longer possible.

**Comment by Mrnofish** [comment]
It is easier, but I don't find the white on black display very soothing, I used to use a dark grey background for eBooks.

> use two -appdata profiles with separate histories

This would be a good idea, however I'd also have to install two separate versions, so that different extensions (ebooks VS pdf) can be associated to the relevant install.

Workable but hardly the greatest, as I would then have to keep both updated, keep settings in sync, and so on.

Is there any planned improvement to bring the new code more in line with the old behavior, or is it just not feasible ?

**Reply by GitHubRulesOK** [reply]
-appdata is primarily for one exe (there are other ways for more than one exe)

You use a folder name for setting file but one can be a sub folder (needs the -appdata for that folder only) of the other and both use the same exe, it is trickier than normal and as you say unsynced overall by that I mean pdfs may be opened in the ePub version and vice versa

The one common MuPDF pdf interface has no colour settings (theory is if a pdf or eBook has a colour set it should be used but neutral whitish otherwise) unless you fiddle background on the fly (was shown a method in the past but it means edit settings.txt via shortcuts using find and replace and could corrupt the file. safer would be an autohotkey or vb script of your own making, but add a backup function.

PDF is a complex beast where there may be no colour for the void that the objects (including scanned images) are rendered on but there are multiple page definitions (boxes) and any of those could be coloured backdrop! using the conventional paper white is the norm.

**Reply by GitHubRulesOK** [reply]
@Mrnofish 
as an example in windows 11 sandbox there is no native image viewer so I can double click an image and get the default settings on the left OR I can have another settings for same exe that I can right click and send images to view, it is same image and exe but different settings

sumatrapdf -appdata C:\Users\WDAGUtilityAccount\Desktop\SandBox\apps\Graphics fox.jpg

![image](https://user-images.githubusercontent.com/12754913/189561495-3322e377-153c-4782-995d-8eb1f9ffde01.png)

Or an ePub same current pre-rel version OR even external view some of those files in secondary version 3.1.2 (still a favorite)

![image](https://user-images.githubusercontent.com/12754913/189561970-725b1116-667e-450b-b626-631098b32077.png)
